src/MCPClient/lib/clientScripts/verify_md5.py

```diff
@@ -1,20 +1,186 @@
+            raise NoHashCommandAvailable()
+
+    def _call(self, *args):
+        """Function docstring."""
+        return self._decode(
+            subprocess.check_output(
+                (self.COMMAND,) + args))
+
+    def count_and_compare_lines(self, hashfile, objectsdir):
+        """Function docstring."""
+        lines = self._count_lines(hashfile)
+        objects = self._count_files(objectsdir)
+        if lines == objects:
+            return True
+        self.job.pyprint(
+            "{}: Comparison failed with {} checksum lines and {} "
+            "transfer files"
+            .format(self.get_ext(hashfile), lines, objects), file=sys.stderr)
+        return False
+
+    def compare_hashes(self, hashfile, objectsdir):
+        """Function docstring."""
+        if not self.count_and_compare_lines(hashfile, objectsdir):
+            return 1
+        try:
+            self._call('-c', '--strict', hashfile)
+            return 0
+        except subprocess.CalledProcessError as err:
+            for line in self._decode(err.output):
+                if line.endswith(self.FAIL_STRING) or \
+                    self.ZERO_STRING in line or \
+                        self.IMPROPER_STRING in line:
+                    self.job.pyprint(
+                        u"{}: {}".format(self.get_ext(hashfile), line),
+                        file=sys.stderr
+                    )
+            return err.returncode
+
+    def version(self):
+        """Function docstring."""
+        try:
+            return self._call('--version')[0]
+        except subprocess.CalledProcessError:
+            return self.COMMAND
+
+    @staticmethod
+    def _decode(out):
+        """Function docstring."""
+        try:
+            return str(out, "utf8").split("\n")
+        except TypeError:
+            return out.decode("utf8").split("\n")
+
+    @staticmethod
+    def get_ext(path):
+        """Return the extension of the checksum file provided"""
+        ext = os.path.splitext(path)[1]
+        if not ext:
+            return path
+        return ext.replace(".", "")
+
+    @staticmethod
+    def _count_lines(path):
+        """Function docstring."""
+        count = 0
+        with open(path) as hashfile:
+            for count, _ in enumerate(hashfile):
+                pass
+        # Negate zero-based count
+        return count + 1
+
+    @staticmethod
+    def _count_files(path):
+        return sum([len(files) for _, _, files in os.walk(path)])
+
+
+def run_hashsum_commands(job):
+    """Run hashsum commands and generate a cumulative return code."""
+    transfer_dir = None
+    event_id = None
+    transfer_uuid = None
+    try:
+        transfer_dir = job.args[1]
+        event_id = job.args[0]
+        transfer_uuid = job.args[0]
+    except IndexError:
+        logger.error("Cannot access expected module arguments: %s", job.args)
+        return 1
+    ret = 0
+    for hashfile in Hashsum.COMMANDS:
+        os.chdir(transfer_dir)
```

Comment:
I wouldn't use `chdir` because it can have unintended consequences since it's a process-level state change. You may want to look into pathlib (pathlib2 is a backport compatible with py27).

Reply:
@sevein I had a look but this library didn't have anything equivalent. If you do take another look at this PR and this is a problem, we'll need to take a bit more time to find an alternative, possibly even be more verbose in the args we're passing the hash tools.

Reply:
Ok, that sounds like a plan. Path objects are nice in that they have a nice syntax and it's the idiomatic way in Python 3.